developer/src/kmc-keyboard/src/compiler/keys.ts

```diff
@@ -71,15 +72,17 @@ export class KeysCompiler extends SectionCompiler {
 
     // Finally, kmap
     // Use LayerMap + keys to generate compiled keys for hardware
-    const theLayers = this.keyboard.layers?.[0]; // TODO-LDML: handle >1 layers. #8160
-
-    if(theLayers?.form == 'hardware') {
+    const hardwareLayers = this.keyboard.layers?.filter(({form}) => form !== 'touch');
+    if (hardwareLayers.length > 1) {
+      // validation should have already caught this
+      throw Error(`Internal error: Expected 0 or 1 hardware layer, not ${hardwareLayers.length}`);
```

Comment:
Are we supporting multiple hardware layers? I thought we were, just not more than one of each form?

Reply:
Briefly How would that work ? Or different widths ?

Reply:
```xml
<layers form="iso">...</layers>
<layers form="us">...</layers>
```

Would give the dev the ability to spec different layouts for 101 vs 102 key keyboards, which could be helpful (e.g. on a 102 key layout can use the extra key more effectively?)

But it is also a bit of a footgun for docs and support.

Reply:
mm this was rejected in _Keyboard-SC February 13/14, 2023_ 

> only a single hardware layout…Avoids confusion since we don't know what's actually plugged in

Reply:
OK